src/webapp/components/form/FormFieldsState.ts

```diff
@@ -195,3 +195,16 @@ export function validateField(
           }
         : undefined;
 }
+
+// RULES:
+
+export function hideFieldsAndSetToEmpty(fields: FormFieldState[]): FormFieldState[] {
+    return fields.map(field => {
+        const fieldWithEmptyValue = getFieldWithEmptyValue(field);
+
+        return {
+            ...fieldWithEmptyValue,
+            isVisible: false,
+        };
```

Comment:
[subjective] [no need to change] That's my TOC: we can write `return fields.map(field => ({ ...getFieldWithEmptyValue(field), isVisible: false }))`, reducing 7 LOCs, with no readability loss (subjective).